core/server/worker/src/main/java/alluxio/worker/block/allocator/RoundRobinAllocator.java

```diff
@@ -70,43 +77,43 @@ public StorageDirView allocateBlockWithView(long sessionId, long blockSize,
    */
   @Nullable
   private StorageDirView allocateBlock(long sessionId, long blockSize,
-      BlockStoreLocation location) {
+      BlockStoreLocation location, boolean skipReview) {
     Preconditions.checkNotNull(location, "location");
     if (location.equals(BlockStoreLocation.anyTier())) {
-      int tierIndex = 0; // always starting from the first tier
       for (int i = 0; i < mMetadataView.getTierViews().size(); i++) {
-        StorageTierView tierView = mMetadataView.getTierViews().get(tierIndex);
+        StorageTierView tierView = mMetadataView.getTierViews().get(i);
+        // The review logic is handled in getNextAvailDirInTier
         int dirViewIndex = getNextAvailDirInTier(tierView, blockSize,
-            BlockStoreLocation.ANY_MEDIUM);
+            BlockStoreLocation.ANY_MEDIUM, skipReview);
         if (dirViewIndex >= 0) {
           mTierAliasToLastDirMap.put(tierView.getTierViewAlias(), dirViewIndex);
           return tierView.getDirView(dirViewIndex);
-        } else { // we didn't find one in this tier, go to next tier
-          tierIndex++;
         }
       }
     } else if (location.equals(BlockStoreLocation.anyDirInTier(location.tierAlias()))) {
       StorageTierView tierView = mMetadataView.getTierView(location.tierAlias());
-      int dirViewIndex = getNextAvailDirInTier(tierView, blockSize, BlockStoreLocation.ANY_MEDIUM);
+      // The review logic is handled in getNextAvailDirInTier
+      int dirViewIndex = getNextAvailDirInTier(tierView, blockSize,
+              BlockStoreLocation.ANY_MEDIUM, skipReview);
       if (dirViewIndex >= 0) {
         mTierAliasToLastDirMap.put(tierView.getTierViewAlias(), dirViewIndex);
         return tierView.getDirView(dirViewIndex);
       }
     } else if (location.equals(BlockStoreLocation.anyDirInAnyTierWithMedium(
-            location.mediumType()))) {
-      String medium = location.mediumType();
-      int tierIndex = 0; // always starting from the first tier
+                location.mediumType()))) {
       for (int i = 0; i < mMetadataView.getTierViews().size(); i++) {
-        StorageTierView tierView = mMetadataView.getTierViews().get(tierIndex);
-        int dirViewIndex = getNextAvailDirInTier(tierView, blockSize, medium);
+        StorageTierView tierView = mMetadataView.getTierViews().get(i);
+        // The review logic is handled in getNextAvailDirInTier
+        int dirViewIndex = getNextAvailDirInTier(tierView, blockSize,
+                location.mediumType(), skipReview);
```

Comment:
I refactored this a bit to remove redundant local variables. I believe the logic are kept the same but please do take an extra look for me. Thanks!